modules/service_accounts/main.tf

```diff
@@ -12,6 +14,45 @@ resource "google_service_account" "main" {
   description  = "Service Account used by Weights & Biases."
 }
 
+# Creates a managed service account key (credentials key)
 resource "google_service_account_key" "main" {
   service_account_id = google_service_account.main.name
-}
\ No newline at end of file
+}
+
+locals {
+  sa_member  = "serviceAccount:${google_service_account.main.email}"
+  project_id = data.google_client_config.current.project
+}
+
+# Service Account Token Creator role allows principals to impersonate
+# service accounts to create OAuth 2.0 tokens which can be used to
+# authenticate with Google APIs 
+resource "google_project_iam_member" "token_creator" {
+  project = local.project_id
+  role    = "roles/iam.serviceAccountTokenCreator"
+  member  = local.sa_member
+}
+
+# PubSub Admin role provides service account members 
+# full access to topics and subscriptions
+resource "google_project_iam_member" "pubsub_admin" {
+  project = local.project_id
+  role    = "roles/pubsub.admin"
+  member  = local.sa_member
+}
+
+# Storage Object Admin role grants service account members
+# full control of objects in a storage bucket
+resource "google_project_iam_member" "storage_object_admin" {
+  project = local.project_id
+  role    = "roles/storage.objectAdmin"
+  member  = local.sa_member
```

Comment:
We should definitely scope this role to the specific bucket.  Same with pubsub.  We really don't need admin either, but it's simplest for now.

Reply:
Updated permissions. The role is now scoped to the specific bucket. Also, updated PubSub role to only bind to the specific topic.